src/engine/Filter.cpp

```diff
@@ -141,7 +156,10 @@ IdTable Filter::computeFilterImpl(
 
   std::visit(visitor, std::move(expressionResult));
 
-  return std::move(output).toDynamic();
+  // Clang 17 seems to incorrectly deduce the type.
+  dynamicResultTable =
+      std::move(const_cast<IdTableStatic<WIDTH>&>(resultTable)).toDynamic();
```

Comment:
Thanks for exploring this, 
I'll maybe clean this up before merging and add a link to the issue, but otherwise this PR now looks really nice.